lua/oil/config.lua

```diff
@@ -323,6 +323,7 @@ local M = {}
 
 ---@class (exact) oil.PreviewWindowConfig
 ---@field update_on_cursor_moved boolean
+---@field win_options? table<string, any>
```

Comment:
You'll want this one to omit the `?` and you'll need to also add this to `oil.SetupPreviewWindowConfig`. This is the internal reference for type safety, the other one is for typing the options passed to `setup()`. It's annoying to keep them both in sync, but it's the only way I could think of to get good type safety (and LuaLS completions) in both locations.

Reply:
Ah that makes sense, thanks for the guidance here. I updated the commit appropriately